### PR TITLE
Homework_7

### PR DESCRIPTION
**Pull request by Vladmaked**

### CODE REVIEW

File: homeworks/vlad.makedon_Vladmaked/homework_7/fibonacci.test.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    });
+
+    test('the second element arrayOfFibonacciNumbers equal one', () => {
+        expect(getFibonacciNumbers()[1]).toBe(1);
+    });
+
+    function getRandomNumber(startOfRange, endOfRange) {
+        return (Math.random() * (endOfRange - startOfRange) + startOfRange);
+    }
+    const rand = getRandomNumber(50, 100);
```
Comment (markoffden) [inline homeworks/vlad.makedon_Vladmaked/homework_7/fibonacci.test.js]: this should be a part of your program, not test for the program

File: homeworks/vlad.makedon_Vladmaked/homework_7/fibonacci.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+function getFibonacciNumbers(rand) {
+    const arrayOfFibonacciNumbers = [0, 1];
+    for (let i = 2; arrayOfFibonacciNumbers[i - 1] + arrayOfFibonacciNumbers[i - 2] < rand; i++) {
```
Comment (markoffden) [inline homeworks/vlad.makedon_Vladmaked/homework_7/fibonacci.js]: I would keep variable naming a bit shorter, something like `fiboArr` in your case

File: homeworks/vlad.makedon_Vladmaked/homework_7/fibonacci.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+        const currentNumber = arrayOfFibonacciNumbers[i - 1] + arrayOfFibonacciNumbers[i - 2];
+        arrayOfFibonacciNumbers.push(currentNumber);
+    }
+    return arrayOfFibonacciNumbers;
+}
```
Comment (markoffden) [inline homeworks/vlad.makedon_Vladmaked/homework_7/fibonacci.js]: also, if you assume that your function `getFibonacciNumbers(rand)` may be called without arguments (as you call it in tests), good practice is to provide some default value for the argument if it is not provided explicitly

File: homeworks/vlad.makedon_Vladmaked/homework_7/fibonacci.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+function getRandomNumber(startOfRange, endOfRange) {
+    return (Math.floor(Math.random() * (endOfRange - startOfRange) + startOfRange));
+}
+
+function getFibonacciNumbers(rand = 55) {
+    const fiboArr = [0, 1];
+    for (let i = 2; fiboArr[i - 1] + fiboArr[i - 2] <= rand; i++) {
+        fiboArr.push(fiboArr[i - 1] + fiboArr[i - 2]);
+    }
```
Comment (markoffden) [inline homeworks/vlad.makedon_Vladmaked/homework_7/fibonacci.js]: cool, but you could do it like `for (let i = 2, let fibo = fiboArr[i - 1] + fiboArr[i - 2]; fibo <= rand; i++) { fiboArr.push(fibo) }`

Comment (Vladmaked) [reply homeworks/vlad.makedon_Vladmaked/homework_7/fibonacci.js]: I wanted to do this, but I did not succeed. I'll play with it again, thanks.

**Pull request merged**